chore(apm/php agent): update compatibility (#21031)
newrelic/docs-website: src/content/docs/apm/agents/php-agent/frameworks-libraries/analyze-phpunit-test-data-new-relic.mdx

freshnessValidatedDate: never
---

> If you use [PHPUnit](https://phpunit.de/) to manage and run your unit tests, the New Relic PHP agent can automatically capture the test summary results and send them to [New Relic](/docs/insights/new-relic-insights/understanding-insights/new-relic-insights) as an [event](/docs/insights/new-relic-insights/understanding-insights/new-relic-insights#database) where you can query and visualize test data at a glance. This feature supports PHPUnit versions <DNT>**8.x**</DNT> only.

Reviewer: ```suggestion
When you use [PHPUnit](https://phpunit.de/) to manage and run your unit tests, the New Relic PHP agent can automatically capture the test summary results. The agent sends this data to the [New Relic platform](/docs/insights/new-relic-insights/understanding-insights/new-relic-insights) as [events](/docs/insights/new-relic-insights/understanding-insights/new-relic-insights#database), allowing you to query, chart, and analyze your test results. This feature supports `PHPUnit` version 8.x and above.
```

Author: @WriteMayur PHP Agent only supports version 8.x of PHPUnit. Your suggestion looks good except for the "and above" in the last sentence.

<Callout variant="caution">

> While we don’t support PHPUnit version 11+, the agent needs to be <DNT>**explicitly disabled**</DNT> by setting `newrelic.enabled` to `false` to avoid out-of-memory issues while running unit tests with PHPUnit 11+. The agent can be disabled while using `phpunit` like this:

Author: The "out-of-memory" errors are not potential, they are certain:
```suggestion
The New Relic PHP agent doesn't support `PHPUnit` version 9 or newer. To prevent out-of-memory errors when running unit tests with `PHPUnit` version 11 or newer, you must <DNT>**explicitly disable**</DNT> the agent by setting `newrelic.enabled` to `false`. You can disable the it while using `phpunit` like this:
```

> While we don’t support PHPUnit version 11+, the agent needs to be <DNT>**explicitly disabled**</DNT> by setting `newrelic.enabled` to `false` to avoid out-of-memory issues while running unit tests with PHPUnit 11+. The agent can be disabled while using `phpunit` like this:

Reviewer: @lavarou, if these is a certain error then lets proceed with your version.

freshnessValidatedDate: never
---

If you use [PHPUnit](https://phpunit.de/) to manage and run your unit tests, the New Relic PHP agent can automatically capture the test summary results and send them to [New Relic](/docs/insights/new-relic-insights/understanding-insights/new-relic-insights) as an [event](/docs/insights/new-relic-insights/understanding-insights/new-relic-insights#database) where you can query and visualize test data at a glance. This feature supports PHPUnit versions <DNT>**8.x**</DNT> only.

<Callout variant="caution">

> Even though New Relic PHP Agent does not support PHPUnit version 11+, there's a known incompatibility between PHPUnit version 11+ and New Relic PHP Agent.
> Therefore we recommend disabling the agent while running unit tests by setting `newrelic.enabled` to `false`. This setting can be passed to `phpunit` like this:

Reviewer 2: ```suggestion
There is a known incompatibility between PHPUnit version 11+ and the New Relic PHP Agent.
While we don't support PHPUnit version 11+, we do recommend disabling the agent to avoid memory issues while running unit tests for PHPUnit 11+ by setting `newrelic.enabled` to `false`. This setting can be passed to `phpunit` like this:
```
